### DISCUSSION TITLE
useShallow for Valtio

### DISCUSSION THREAD

**Discussion by kirankunigiri** [pmndrs/valtio#871] [Ideas]
I used Zustand for several projects, and decided to give Valtio a try. I'm really enjoying it so far because of the mutable state model, but I'm missing the useShallow feature from Zustand. Being able to avoid re-renders based on whether the result of a computation changes or not was extremely handy. Is there a way to do this in Valtio?

Here is a specific example I am dealing with. I have several items, and I want them to re-render when they are clicked on and selected.
In Zustand, with useShallow, the component will only re-render if isSelected changes, instead of every time selectedItem changes.
```ts
const isSelected = useMainStore(useShallow((state) => state.selectedItem === props.item.id));
```
How would I do this in Valtio?

**Comment by dai-shi** [comment]
By the way, you don't need `useShallow` in Zustand,

```js
const isSelected = useMainStore((state) => state.selectedItem === props.item.id);
```

is fine. Don't overuse `useShallow`.

**Comment by dai-shi** [comment]
This should help: https://github.com/pmndrs/valtio/blob/9593184ad94928e54382bfd0e0beb7d1416eb6fe/docs/how-tos/how-to-avoid-rerenders-manually.mdx#L29-L46
